cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -483,13 +491,18 @@ class GroupByNode : public ExecNode {
     return Status::OK();
   }
 
-  void InputReceived(ExecNode* input, ExecBatch batch) override {
+  void InputReceived(ExecNode* input, std::function<Result<ExecBatch>()> task) override {
     // bail if StopProducing was called
     if (finished_.is_finished()) return;
 
     DCHECK_EQ(input, inputs_[0]);
 
-    if (ErrorIfNotOk(Consume(std::move(batch)))) return;
+    auto prev = task();
+    if (!prev.ok()) {
+      ErrorIfNotOk(prev.status());
+      return;
+    }
+    if (ErrorIfNotOk(Consume(prev.MoveValueUnsafe()))) return;
```

Comment:
```suggestion
    auto prev = task();
    if (!prev.ok()) {
      ErrorIfNotOk(prev.status());
      return;
    }
    if (ErrorIfNotOk(Consume(prev.MoveValueUnsafe()))) return;
```

This should change to submit task

Reply:
I would consider to change this once we have ready the interfaces for scheduler and submit task, please note again that this PR is not for defining those interfaces yet, but to prepare the ground and make the input nodes behave more like task generators (for the future scheduler)